src/Venturecraft/Revisionable/RevisionableTrait.php

```diff
@@ -125,7 +125,7 @@ public function preSave()
             // we can only safely compare basic items,
             // so for now we drop any object based items, like DateTime
             foreach ($this->updatedData as $key => $val) {
-                if (isset($this->casts[$key]) && in_array($this->casts[$key], ['object', 'array']) && isset($this->originalData[$key])) {
+                if (gettype($val) == 'object' && isset($this->casts[$key]) && in_array($this->casts[$key], ['object', 'array']) && isset($this->originalData[$key])) {
```

Comment:
This will fail if the data is cast to an `array`

Reply:
Hm, that's interesting.  The attribute I tested with is cast as an array, but this still works.  Maybe internally Laravel still treats it as an object?  I'll update the pull request to check for either type.

Reply:
Oh, I bet it's because I used an associative array.